test/test_docling_doc.py

```diff
@@ -1248,3 +1257,125 @@ def test_save_to_disk():
     _verify_saved_output(filename=filename, paths=paths)
 
     assert True
+
+
+def test_document_stack_operations():
+
+    doc: DoclingDocument = _construct_doc()
+
+    # _print(document=doc)
+
+    ref = RefItem(cref="#/texts/12")
+    success, stack = doc._get_stack_of_refitem(ref=ref)
+
+    assert success
+    assert stack == [
+        2,
+        2,
+        2,
+        0,
+        2,
+        0,
+        0,
+    ], f"stack==[2, 2, 2, 0, 2, 0, 0] for stack: {stack}"
+
+
+def test_document_manipulation():
```

Comment:
Still uncovered cases:
- `replace_item()`
- error case of adding a sibling to the root (body), which should raise an exception
- error case of adding a child which itself has children